---
layout: single
tags: Snaketool, Snakemake, phage, assembly
title: Sphae
author: Bhavya Papudeshi
logo: /assets/images/sphae_logo.png
---

![](/assets/images/sphae_logo.png)

[![Edwards Lab](https://img.shields.io/badge/Bioinformatics-EdwardsLab-03A9F4)](https://edwards.flinders.edu.au)
[![DOI](https://zenodo.org/badge/403889262.svg)](https://zenodo.org/doi/10.5281/zenodo.8365088)
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)

![GitHub language count](https://img.shields.io/github/languages/count/linsalrob/spae)
[![](https://img.shields.io/static/v1?label=CLI&message=Snaketool&color=blueviolet)](https://github.com/beardymcjohnface/Snaketool)
![GitHub last commit (branch)](https://img.shields.io/github/last-commit/linsalrob/spae/main)
[![CI](https://github.com/linsalrob/spae/actions/workflows/testing.yml/badge.svg)](https://github.com/linsalrob/spae/actions/workflows/testing.yml)

[![install with pip](https://img.shields.io/static/v1?label=Install%20with&message=PIP&color=success)](https://pypi.org/project/sphae/)
[![Pip Downloads](https://static.pepy.tech/badge/sphae)](https://www.pepy.tech/projects/sphae)
[![install with bioconda](https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat)](http://bioconda.github.io/recipes/sphae/README.html)
[![Bioconda Downloads](https://img.shields.io/conda/dn/bioconda/sphae)](https://img.shields.io/conda/dn/bioconda/sphae)
![Docker Pulls](https://img.shields.io/docker/pulls/npbhavya/sphae.svg)

Phage toolkit to detect and assemble phage candidates for phage therapy.<!--more--> Snakemake workflow using Snaketool to assemble and annotate phage sequences.<!--more-->

 - [Sphae GitHub Homepage](https://github.com/linsalrob/sphae)

This snakemake workflow was built using [Snaketool](https://github.com/beardymcjohnface/Snaketool), to assemble and annotate phage sequences. Currently this tool is being developed for phage genomes. The steps include

 - QC using [Trimanammi](https://github.com/beardymcjohnface/Trimnami)
 - Assembly, either [SPAdes](https://github.com/ablab/spades) or [Flye](https://github.com/fenderglass/Flye)
 - Contig quality checks
   - Read coverage using [Koverage](https://github.com/beardymcjohnface/Koverage)
   - Verify they are viral contigs using [ViralVerify](https://github.com/ablab/viralVerify)
   - Completeness using [CheckV](https://bitbucket.org/berkeleylab/CheckV)
   - Assembly graph compnents check - internal script This results in determining if the phage genome was assembled
 - Annotation of the phage genome using [Pharokka](https://github.com/gbouras13/pharokka)
