---
layout: single
tags: Snaketool, Snakemake, read trimming, qc
title: Trimnami
author: Michael Roach
---

Trim lots of sequencing reads all at once!<!--more-->

![](/assets/images/trimnami.png)

[![](https://img.shields.io/static/v1?label=CLI&message=Snaketool&color=blueviolet)](https://github.com/beardymcjohnface/Snaketool)
[![](https://img.shields.io/static/v1?label=Licence&message=MIT&color=black)](https://opensource.org/license/mit/)
[![](https://img.shields.io/static/v1?label=Install%20with&message=PIP&color=success)](https://pypi.org/project/trimnami/)
![GitHub last commit (branch)](https://img.shields.io/github/last-commit/beardymcjohnface/Trimnami/main)
[![Unit tests](https://github.com/beardymcjohnface/Trimnami/actions/workflows/python-app.yml/badge.svg)](https://github.com/beardymcjohnface/Trimnami/actions/workflows/python-app.yml)
[![codecov](https://codecov.io/gh/beardymcjohnface/Trimnami/branch/main/graph/badge.svg?token=E0w8zHLLDq)](https://codecov.io/gh/beardymcjohnface/Trimnami)

We keep writing pipelines that start with read trimming. 
Rather than copy-pasting code each time, this standalone [Snaketool](https://github.com/beardymcjohnface/Snaketool) 
handles all our trimming needs. The tool will collect sample names and files from a directory or TSV file, optionally 
remove host reads, and trim with your favourite read trimmer. Refer to the GitHub page for complete list of supported trimmers.

 - [Trimnami GitHub homepage](https://github.com/beardymcjohnface/Trimnami)

Trimnami utilises [MetaSnek](https://github.com/beardymcjohnface/metasnek) for parsing fastq reads and is able to 
correctly identify all paried and single end sequencing files, from either a directory or TSV sample manifest input.
In addition to read-trimming, Trimnami can also generate multi-fastqc report for the trimmed and untrimmed files,
perform subsampling on the trimmed reads, perform host contaminant removal with a user-supplied genome, and Trimnami
supports both FASTQ and FASTA output formats. Trimnami is designed for interoperability with other Snaketool and 
Snakemake pipelines and has been incorporated into numerous FAME pipelines such as 
[Hecatomb]()https://github.com/shandley/hecatomb, and [Spae](https://github.com/linsalrob/spae).